cartridge-cli.lua

```diff
@@ -2832,13 +3048,74 @@ local function instantiate_template(template_dir, dest_dir, app_name)
         local destdir = fio.dirname(destname)
 
         if not fio.path.exists(destdir) then
-            make_tree(destdir)
+            local ok, err = make_tree(destdir)
+            if not ok then return false, err end
         end
 
-        write_file(destname, content, mode)
+        local ok, err = write_file(destname, content, mode)
+        if not ok then return false, err end
     end
+
+    return true
 end
 
+local function create_app_directory_and_init_git(dest_dir, template, name)
+    assert(fio.path.exists(dest_dir))
+
+    local template_dir = fio.pathjoin(get_template_dir(), template)
+
+    if not fio.path.exists(template_dir) then
+        return false, string.format("Template '%s' doesn't exist", template_dir)
+    end
+
+    local ok, err = instantiate_template(template_dir, dest_dir, name)
+    if not ok then
+        return false, string.format('Failed to instantiate application template: %s', err)
+    end
+
+    local git = which('git')
+
+    if git ~= nil then
+        info("Initializing git repo in: %s", dest_dir)
+
+        repeat  -- until true
+            local ok, err = call("cd %s && %s init .", dest_dir, git)
+            if not ok then
+                warn('Failed to initialize git repo: %s', err)
+                break
```

Comment:
Why not simple `return`? 

Why `break`?

Reply:
If some additional actions will be added after git init, it will require changing all these `break` keywords to `return`.